zeppelin-server/src/main/java/org/apache/zeppelin/server/AppScriptServlet.java

```diff
@@ -85,7 +83,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     int endIndex = script.indexOf(endReplaceString, startIndex);
 
     if (startIndex >= 0 && endIndex >= 0) {
-      String replaceString = "this.getPort=function(){return " + websocketPort + "};";
+      String replaceString = "this.getPort=function(){return "
+          + ZeppelinServer.jettyServer.getConnectors()[0].getLocalPort() + "};";
```

Comment:
This will set an incorrect internal local port of the zeppelin instance (e.g. 8080) where it is possible that there is a reverse proxy (nginx with HTTPS for example) in front of zeppelin. 

Actually I think this whole Servlet is no longer needed since the websocket port is now the http port, so the browser can simply assume location.port